File: leetcode3/염혜정/3992. Rearrange String to Avoid Character Pair.java
// O(n)

class Solution {
    public String rearrangeString(String s, char x, char y) {
        StringBuilder left = new StringBuilder();
        StringBuilder mid = new StringBuilder();
        StringBuilder right = new StringBuilder();

        for (char c : s.toCharArray()) {
            if (c == y) left.append(c);
            else if (c == x) right.append(c);
            else mid.append(c);
        }
        return left.toString() + mid.toString() + right.toString();
    }
}

File: leetcode3/변지협/v2/3992. Rearrange String to Avoid Character Pair.py

'''
1. 아이디어 :
그냥 y를 먼저 붙이고, 나머지 문자를 붙인다.
x 몰라도 그냥 풀림

2. 시간복잡도 :
o(n)

3. 자료구조/알고리즘 :
'''

class Solution:
    def rearrangeString(self, s: str, x: str, y: str) -> str:
        ans = ''
        tmp = ''
        
        for i in s:
            if i == y:
                tmp += i
            else:
                ans += i
        
        return tmp+ans